### PR TITLE
Sync actions product assets

### LINKED ISSUE #1385: No support for assets in product syncActions
As titled.

### PR DESCRIPTION
**Pull request by katmatt**
#### Summary

This PR implements support for product variant assets in the `sync-actions` module.

Closes #1385

#### Todo

- Tests
  - [x] Unit
  - [ ] Integration
  - [ ] Acceptance
- [ ] Documentation

### CODE REVIEW

**Comment by katmatt** [comment]
@daern91 Looks like some of our integration tests are failing, but as far as i can see my new tests are running successfully 😁

**Comment by daern91** [comment]
Hi @katmatt ,

Thanks a ton for the PR! I only see that the linter is failing but it doesn't seem to have anything to do with this PR... Hmm, I can look into it.

**Comment by katmatt** [comment]
@daern91 Now everything should work, would be great if you could review it 😄

File: packages/sync-actions/src/product-actions.js [hunk 1]
```diff
@@ -366,6 +367,51 @@ function _buildVariantAttributesActions(
...
+    }),
+    [REMOVE_ACTIONS]: oldAsset => ({
+      action: 'removeAsset',
+      ...toAssetIdentifier(oldAsset),
+      ...toVariantIdentifier(oldObj),
+    }),
+    [CHANGE_ACTIONS]: (oldAsset, newAsset) =>
+      // here we could use more atomic update actions (e.g. changeAssetName)
+      // but for now we use the simpler approach to first remove and then
+      // re-add the asset - which reduces the code complexity
```
Comment (daern91) [inline packages/sync-actions/src/product-actions.js]: This sounds good to me 👍

**Comment by StarpTech** [comment]
What I often miss is a simple example in the docs. I don't want to run through code in enterprise software :smile:

**Pull request merged**